### DISCUSSION TITLE
Implementation for Göteborgs stad

### DISCUSSION THREAD

**Discussion by viktorlarsson** [kolplattformen/embedded-api#134] [General]
* How do we select the city (Stockholm, Gothenburg) without breaking and scaling this implementation to several other cities?

**Comment by ErikHellman** [comment]
One challenge with embedded-api is that we can't link the project into the skolplattformen project but have to install it as an npm package (a limitation in React Native tool chain). 

I think we should go for a mono-repo approach for the entire app. However, we should probably move the website to its own repo as well. 

We move the code for embedded-api into the skolplattformen app project and place it under a "stockholm" source folder. Then we create another folder for each new integration. 

The user would choose a city on the first launch and we store the choice locally (with the option to change it later).

**Comment by JohanObrink** [comment]
@ErikHellman That is probably a good idea. I am currently migrating a monorepo from lerna to Nx which I think I definitely prefer since it relies on config over magic. We should try setting up an Nx workspace, creating libs for api-stockholm and hooks and an app for Skolplattformen and then moving the existing code into it.

Step 2 should be to separate types and interfaces from embedded-api and into a common dependency of all api implementations.
